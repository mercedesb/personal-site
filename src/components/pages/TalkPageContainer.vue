<template>
   <TalkPage :page="page" />
</template>

<script>
import TalkPage from './TalkPage.vue'
import objects from '../../mixins/objects'
import images from '../../mixins/images'

const moment = require('moment')

export default {
  components: {
    TalkPage
  },
  mixins: [
    objects, images
  ],
  metaInfo () {
    return {
      title: this.page.title,
      titleTemplate: '%s | Mercedes Bernard',
      meta: [
        { name: 'description', vmid: 'description', content: this.page.preamble },
        { name: 'og:title', vmid: 'og:title', content: this.page.title },
        // Twitter Card data
        { name: 'twitter:card', vmid: 'twitter:card', content: 'summary' },
        { name: 'twitter:site', vmid: 'twitter:site', content: '@mercedescodes' },
        { name: 'twitter:title', vmid: 'twitter:title', content: this.page.title },
        { name: 'twitter:description', vmid: 'twitter:description', content: this.page.preamble },
        { name: 'twitter:creator', vmid: 'twitter:creator', content: '@mercedescodes' },
        // Open Graph data
        { property: 'og:title', vmid: 'og:title', content: this.page.title },
        { property: 'og:type', vmid: 'og:type', content: 'article' },
        { property: 'og:url', vmid: 'og:url', content: `http://www.mercedesbernard.com${this.$route.fullPath}/` },
        { property: 'og:description', vmid: 'og:description', content: this.page.preamble },
        { property: 'og:site_name', vmid: 'og:site_name', content: 'Mercedes Bernard' }
      ]
    }
  },
  computed: {
    page () {
      const talk = this.$store.state.talk
      let givenAt = []
      if (talk && talk.givenAt) {
        givenAt = talk.givenAt.map(ga => {
          return {
            id: ga.sys.id,
            ...ga.fields,
            date: moment(ga.fields.date).format('MMMM DD, YYYY'),
            slidesLink: this.getImageUrl(ga.fields.slidesLink)
          }
        })
      }
      return {
        ...talk,
        iconSvg: talk.iconSvg ? talk.iconSvg.fields.svg : null,
        givenAt: givenAt
      }
    }
  },
  created () {
    this.$store.dispatch('getTalk', this.$route.params.urlSegment)
      .then((page) => {
        if (this.isEmpty(page)) {
          this.$router.push({name: 'pageNotFound'})
        }

        this.$store.dispatch('getNavLinks')
          .then(() => {
            document.dispatchEvent(new Event('custom-render-trigger'))
          })
      })
  }
}
</script>