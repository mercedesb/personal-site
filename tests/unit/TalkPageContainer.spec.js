import TalkPageContainer from "@/components/pages/TalkPageContainer.vue";
import { TestUtility } from "./TestUtility";

const mockFormat = jest.fn(() => "Apr 18 2018");

jest.mock("moment", () => () => ({
  format: mockFormat
}));

describe("TalkPageContainer", () => {
  let component;

  const store = {
    state: {
      talk: TestUtility.talks[0],
      navLinks: []
    }
  };

  const shallow = propsData =>
    TestUtility.shallow(TalkPageContainer, {
      store,
      propsData: {
        ...propsData
      }
    });

  describe("Snapshots", () => {
    it("matches snapshot", () => {
      component = shallow();
      expect(component.element).toMatchSnapshot();
    });
  });

  describe("Computed", () => {
    describe("page", () => {
      beforeEach(() => {
        component = shallow();
      });

      it("has page populated", () => {
        expect(component.vm.page).toBeDefined();
      });

      it("formats the given at date", () => {
        expect(typeof component.vm.page.givenAt[0].date).toBe("string");
      });
    });
  });

  describe("Lifecycle", () => {
    describe("created", () => {
      it("dispatches getTalk to the store", () => {
        component = TestUtility.mount(TalkPageContainer, {
          store
        });
        const expectedPath = TestUtility.mockRoute;
        expect(component.vm.$store.dispatch).toHaveBeenCalledWith(
          "getTalk",
          expectedPath
        );
        // TODO: test getNavLinks
      });
    });
  });
});
