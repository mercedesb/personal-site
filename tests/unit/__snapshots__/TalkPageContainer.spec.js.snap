// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TalkPageContainer Snapshots matches snapshot 1`] = `
<talkpage-stub
  page="[object Object]"
/>
`;
